src/charm.py

```diff
@@ -341,48 +339,41 @@ def _config_changed(self, event):
         # First execute the setup script in 2 conditions:
         # - First run (when no services are planned in pebble)
         # - Change in important S3 parameter (comparing value with envVars in pebble plan)
-        script = f"{SCRIPT_PATH}/pod_setup"
         if (
             self._is_config_valid()
             and self.model.unit.is_leader()
             and self._should_run_setup(current_plan, previous_s3_info)
         ):
-
-            logger.debug("Executing setup script (%s)", script)
+            self.model.unit.status = MaintenanceStatus("Compiling assets")
+            script = f"{SCRIPT_PATH}/pod_setup"
             process = container.exec(
-                [script], environment=self._create_discourse_environment_settings()
+                [script],
+                environment=self._create_discourse_environment_settings(),
+                working_dir="/srv/discourse/app",
             )
             try:
-                self.model.unit.status = MaintenanceStatus(f"Executing setup {script}")
                 stdout, _ = process.wait_output()
                 logger.debug("%s stdout: %s", script, stdout)
             except ExecError as e:
                 logger.error("%s command exited with code %d. Stderr:", script, e.exit_code)
                 for line in e.stderr.splitlines():
                     logger.error("    %s", line)
-                logger.debug("%s stdout: %s", script, e.stdout)
-                self.model.unit.status = BlockedStatus(f"Error while executing {script}")
```

Comment:
Will the charm put in BlockedState itself in case of Exception ?
Is so, what will be the message shown ? The idea here was to have a self-explaining message in case of BlockedState